src/domain_classes/dimension.py

```diff
@@ -3,53 +3,31 @@
 
 class Dimension:
     def __init__(self, dimensions: str, attribute_type: str):
+        """
+        dimensions: define the structure of the array. For example:
+        - "*" is a one dimensional array of any length.
+        - "*,*" is a two dimensional array of any length.
+        - "2,1" is a 2 dimensional array where inner arrays have length one - for example [[1], [3]]
+
+        type: define what type the values in the array have. Primitive types (bool, int, str, etc)
+        are converted to python types. For complex types, for example "dmss://system/SIMOS/Package",
+        a string value is stored.
+        """
         self.dimensions: list[str] = dimensions.split(",")
-        self.type: str = get_data_type_from_dmt_type(attribute_type)
+        self.type: type[bool] | type[int] | type[float] | type[str] | str = get_data_type_from_dmt_type(attribute_type)
         self.value = None
 
-    def is_array(self):
+    def is_array(self) -> bool:
         return self.dimensions != [""]
 
-    def is_matrix(self):
+    # TODO remove function - it is never used for anything useful
```

Comment:
Or rewrite to `get_dimensions(self) -> int`?

Reply:
since I didn't see any apparent uses of get_dimensions() (without too much investigation), decided to skip this one.